### ISSUE TITLE
Call uninstall target before install

### ISSUE THREAD

**Issue by drdanz** [robotology/ycm#29]
In some cases it would be nice to have "make uninstall" called before "make install" (or better even before update and build steps). This would allow to have a clean situation before the installation and would avoid to have files that are no longer installed in the install directory.

An example of issue caused by not running uninstall before install is this: robotology/yarp#159

**Comment by DanielePucci** [comment]
I agree! I was having some troubles after the update of `IOpenLoopControl` and since `icub-main` was not compiling. Then @traversaro suggested me to clean the `install` directory and everything worked: a proper `make uninstall` would have done the job for me :)

**Comment by drdanz** [comment]
Just adding this before any `install` command does the trick:

```cmake
install(SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
```
